packages/examples/bike_simulator.py

```diff
@@ -38,9 +38,14 @@
 
 class PackageClass(LiotaPackage):
 
-    def run(self, registry):
+    def run(self, registry, package_record):
         from liota.entities.devices.bike_simulated import BikeSimulated
         import pint
+        from liota.lib.utilities.utility import check_integrity
+
+        # verify file integrity first
+        if (check_integrity(package_record, __file__) == False):
```

Comment:
You are correct. Thanks for pointing it out.

This check should actually be done in package_manager near the beginning of _package_load method right after calculating the file hash. The original hash passed through the pipe should be stored and be compared with this hash. An alternative would be to place the check in _load_module_package.